### PR TITLE
feat(processors): Add baggage span processor

### PR DESCRIPTION
**Pull request by martin308**
## Short description of the changes

Add a new processor that copies the current baggage as attributes to the child span

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208116694462800

### CODE REVIEW

File: Tests/Honeycomb/ReadableSpanMock.swift [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+@testable import OpenTelemetrySdk
+
+class ReadableSpanMock: ReadableSpan {
+    var hasEnded: Bool = false
+    var latency: TimeInterval = 0
+
+    var kind: SpanKind {
+        return .client
+    }
```
Comment (beekhc) [inline Tests/Honeycomb/ReadableSpanMock.swift:15]: can this be written as

`var kind: SpanKind { .client }`

**Review by beekhc — CHANGES_REQUESTED** [review]
This change looks good, and the tests look good!

I just have a bunch of little nits about swift style.

File: Tests/Honeycomb/ReadableSpanMock.swift (hunk 1 above)
Comment (martin308) [reply Tests/Honeycomb/ReadableSpanMock.swift:15]: appears so, will change

File: Tests/Honeycomb/HoneycombBaggageSpanProcessorTests.swift [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import OpenTelemetrySdk
+import XCTest
+
+@testable import Honeycomb
+
+class HoneycombBaggageSpanProcessorTests: XCTestCase {
+    let readableSpan = ReadableSpanMock()
+
+    func testNoCrash() {
+        let processor = HoneycombBaggageSpanProcessor(filter: { (e) in return true })
```
[the remark this answers is not in the document]
Comment (martin308) [reply Tests/Honeycomb/HoneycombBaggageSpanProcessorTests.swift:12]: Xcode seems to think it needs to look like `{ _ in true }`, which is def a little shorter. Does that seem right?

File: Tests/Honeycomb/ReadableSpanMock.swift [hunk 3]
```diff
@@ -0,0 +1,78 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+import Foundation
+import OpenTelemetryApi
+
+@testable import OpenTelemetrySdk
```
[the remark this answers is not in the document]
Comment (martin308) [reply Tests/Honeycomb/ReadableSpanMock.swift:7]: unfortunately we need to construct `SpanData` and we're unable to without using the internal init function

**Review by beekhc — APPROVED** [review]
looks great 🚀

**Pull request merged**